mlx/export.h

```diff
@@ -0,0 +1,24 @@
+// Copyright © 2024 Apple Inc.
+
+#pragma once
+
+#include "mlx/array.h"
+
+namespace mlx::core {
+
+/**
+ * Export a function to a file.
+ */
+void export_function(
+    std::string path,
+    const std::function<std::vector<array>(const std::vector<array>&)>& fun,
+    const std::vector<array>& inputs,
+    bool shapeless = false);
+
+/**
+ * Import a function from a file.
+ */
+std::function<std::vector<array>(const std::vector<array>&)> import_function(
+    std::string path);
```

Comment:
Another question: should `import_function` return metadata? I can see how it would be useful to get say the shapes and/or dtypes of the inputs, maybe the MLX version, etc in a dict of metadata. Can also wait and see and provide an overload / a `return_metadata` flag in the future.

Reply:
I think it's fine. I am not too sure why we want to put everything in a single file tbh (weights, metadata and computation graph) but either way, as you say, we can always add a `return_metadata` optional argument.